descartes_light/solvers/include/descartes_light/solvers/bgl/impl/bgl_dijkstra_solver.hpp

```diff
@@ -102,140 +47,78 @@ SearchResult<FloatType> BGLEfficientDijkstraSVSESolver<FloatType>::search()
                                    static_cast<FloatType>(0.0),
                                    visitor,
                                    color_prop_map);
+
+    // In the case that the visitor does not throw the target vertex descriptor, find the lowest cost vertex in last
+    // rung of the ladder graph
+    auto target = std::min_element(ladder_rungs.back().begin(),
+                                   ladder_rungs.back().end(),
+                                   [&](const VertexDesc<FloatType>& a, const VertexDesc<FloatType>& b) {
+                                     return graph[a].distance < graph[b].distance;
+                                   });
+    if (target != ladder_rungs.back().end())
+      throw *target;
```

Comment:
Could this prevent a failed search from failing? I would think this causes he catch below to always trigger

Reply:
I think you're right. I think I need to add a check to see whether the cost of that vertex is also less than the float max